src/storage/storage.ts

```diff
@@ -141,6 +141,25 @@ export class Storage implements IStorage {
     return this.store.getCurrentAddress(markAsUsed);
   }
 
+  /**
+   * Get a change address to use, if one is provided we need to check we own its
+   * If not provided the current address will be used instead.
+   *
+   * @param {Object} [options={}]
+   * @param {string|null|undefined} [options.changeAddress=undefined] User provided change address to use
+   * @returns {Promise<string>} The change address to use
+   */
+  async getChangeAddress({ changeAddress }: { changeAddress?: string | null | undefined; } = {}): Promise<string> {
+    if (changeAddress) {
```

Comment:
As a suggestion for [flattening the code](https://refactoring.guru/replace-nested-conditional-with-guard-clauses) with guard clauses:
```ts
if (!changeAddress) {
  return getCurrentAddress();
}

if (!isAddressMine) {
  throw "Not from wallet"
}

return changeAddress;
```